Assertion failed 'value != 0' in 'System.Tests.DoubleTests_GenericMath:GetExponentShortestBitLengthTest()' during 'Do value numbering'

```
ILC: Assertion failed 'value != 0' in 'System.Tests.DoubleTests_GenericMath:GetExponentShortestBitLengthTest()' during 'Do value numbering' (IL size 314; hash 0x60919730; FullOpts)
  
D:\a\_work\1\s\artifacts\bin\coreclr\windows.x64.Checked\build\Microsoft.NETCore.Native.targets(262,5): error MSB3073: The command ""D:\a\_work\1\s\artifacts\bin\coreclr\windows.x64.Checked\ilc-published\\ilc" @"D:\a\_work\1\s\artifacts\obj\System.Runtime.Tests\Release\net8.0-windows\native\System.Runtime.Tests.ilc.rsp"" exited with code 57005. [D:\a\_work\1\s\src\libraries\System.Runtime\tests\System.Runtime.Tests.csproj::TargetFramework=net8.0-windows]
##[error]artifacts\bin\coreclr\windows.x64.Checked\build\Microsoft.NETCore.Native.targets(262,5): error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Build) The command ""D:\a\_work\1\s\artifacts\bin\coreclr\windows.x64.Checked\ilc-published\\ilc" @"D:\a\_work\1\s\artifacts\obj\System.Runtime.Tests\Release\net8.0-windows\native\System.Runtime.Tests.ilc.rsp"" exited with code 57005.

```

This is failing the NativeAOT System.Runtime tests in runtime-extra-platform runs with checked runtime/release framework combo.

https://dev.azure.com/dnceng-public/public/_build/results?buildId=155217&view=logs&jobId=62c84a0c-1188-56fe-6938-e058a633beb1&j=62c84a0c-1188-56fe-6938-e058a633beb1&t=e508e8f5-7a40-5591-4aae-9625248f10c0

Looks to be related to #80789 cc @tannergooding

--- a/src/coreclr/jit/valuenum.cpp
+++ b/src/coreclr/jit/valuenum.cpp
@@ -6121,7 +6121,7 @@ ValueNum ValueNumStore::EvalHWIntrinsicFunUnary(
             case NI_X86Base_BitScanForward:
             {
                 assert(!varTypeIsSmall(type) && !varTypeIsLong(type));
-                int32_t value  = GetConstantInt32(arg0VN);
+                int32_t value = GetConstantInt32(arg0VN);
 
                 if (value == 0)
                 {

--- a/src/coreclr/jit/valuenum.cpp
+++ b/src/coreclr/jit/valuenum.cpp
@@ -6121,40 +6121,60 @@ ValueNum ValueNumStore::EvalHWIntrinsicFunUnary(
             case NI_X86Base_BitScanForward:
             {
                 assert(!varTypeIsSmall(type) && !varTypeIsLong(type));
+                int32_t value  = GetConstantInt32(arg0VN);
 
-                int32_t  value  = GetConstantInt32(arg0VN);
+                if (value == 0)
+                {
+                    // bsf is undefined for 0
+                    break;
+                }
+
                 uint32_t result = BitOperations::BitScanForward(static_cast<uint32_t>(value));
-
                 return VNForIntCon(static_cast<int32_t>(result));
             }
 
             case NI_X86Base_X64_BitScanForward:
             {
                 assert(varTypeIsLong(type));
+                int64_t value = GetConstantInt64(arg0VN);
 
-                int64_t  value  = GetConstantInt64(arg0VN);
+                if (value == 0)
+                {
+                    // bsf is undefined for 0
+                    break;
+                }
+
                 uint32_t result = BitOperations::BitScanForward(static_cast<uint64_t>(value));
-
                 return VNForLongCon(static_cast<int64_t>(result));
             }
 
             case NI_X86Base_BitScanReverse:
             {
                 assert(!varTypeIsSmall(type) && !varTypeIsLong(type));
+                int32_t value = GetConstantInt32(arg0VN);
 
-                int32_t  value  = GetConstantInt32(arg0VN);
+                if (value == 0)
+                {
+                    // bsr is undefined for 0
+                    break;
+                }
+
                 uint32_t result = BitOperations::BitScanReverse(static_cast<uint32_t>(value));
-
                 return VNForIntCon(static_cast<int32_t>(result));
             }
 
             case NI_X86Base_X64_BitScanReverse:
             {
                 assert(varTypeIsLong(type));
+                int64_t value = GetConstantInt64(arg0VN);
 
-                int64_t  value  = GetConstantInt64(arg0VN);
+                if (value == 0)
+                {
+                    // bsr is undefined for 0
+                    break;
+                }
+
                 uint32_t result = BitOperations::BitScanReverse(static_cast<uint64_t>(value));
-
                 return VNForLongCon(static_cast<int64_t>(result));
             }
 #endif // TARGET_XARCH

--- a/src/tests/JIT/Regression/JitBlue/Runtime_81460/Runtime_81460.cs
+++ b/src/tests/JIT/Regression/JitBlue/Runtime_81460/Runtime_81460.cs
@@ -1,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+public static class FloatingPointHelper<TSelf>
+    where TSelf : IFloatingPoint<TSelf>
+{
+    public static int GetExponentShortestBitLength(TSelf value)
+        => value.GetExponentShortestBitLength();
+}
+
+class Runtime_81460
+{
+    static void Main() => Test();
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static int Test() =>
+        FloatingPointHelper<double>.GetExponentShortestBitLength(1.0);
+}

--- a/src/tests/JIT/Regression/JitBlue/Runtime_81460/Runtime_81460.cs
+++ b/src/tests/JIT/Regression/JitBlue/Runtime_81460/Runtime_81460.cs
@@ -1,6 +1,10 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+
 public static class FloatingPointHelper<TSelf>
     where TSelf : IFloatingPoint<TSelf>
 {
@@ -10,7 +14,10 @@ public static class FloatingPointHelper<TSelf>
 
 class Runtime_81460
 {
-    static void Main() => Test();
+    static int Main()
+    {
+        return (Test() == 0) ? 100 : 0;
+    }
 
     [MethodImpl(MethodImplOptions.NoInlining)]
     public static int Test() =>

--- a/src/tests/JIT/Regression/JitBlue/Runtime_81460/Runtime_81460.csproj
+++ b/src/tests/JIT/Regression/JitBlue/Runtime_81460/Runtime_81460.csproj
@@ -1,0 +1,9 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <Optimize>True</Optimize>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="$(MSBuildProjectName).cs" />
+  </ItemGroup>
+</Project>

